strax/config.py

```diff
@@ -116,7 +117,13 @@ def __init__(self,
             raise RuntimeError(f"Tried to specify more than one default "
                                f"for option {self.name}.")
 
-        if type is OMITTED and default is not OMITTED:
+        if infer_dtype is OMITTED and type is OMITTED and default is not OMITTED:
+            warnings.warn(f'You are setting a default value for config {name} but not \
+                specifying a type. In the future the type will be inferred from \
+                the default value which will result in an error if this config \
+                is set to a different type.')
+
+        if infer_dtype and type is OMITTED and default is not OMITTED:
```

Comment:
```suggestion
        if infer_dtype is OMITTED and type is OMITTED and default is not OMITTED:
            warnings.warn(f'You are setting a default value for config {name} but not \
                specifying a type. In the future the type will be inferred from \
                the default value which will result in an error if this config \
                is set to a different type.')

        elif type is OMITTED and default is not OMITTED:
```

Reply:
if i set `infer_dtype=False` wouldnt this still infer the type?

Reply:
fair but `if infer_dtype` is `True` because it is a string, so we would get both a warning and later an error in the validate part.

Reply:
just edited the suggestion

Reply:
oops i pushed an edit before i noticed you also did...

Reply:
either is fine with me.

Reply:
yeah I think this many if statements are indeed a bit unnecessarily tars to follow but your method is good 👍